embeddings/metric/lightning_seqeval_metric.py

```diff
@@ -0,0 +1,77 @@
+from typing import Dict, List, Optional
+
+import torch
+from datasets import ClassLabel
+from torchmetrics import Metric
+
+from embeddings.evaluator.sequence_labeling_evaluator import (
+    EvaluationMode,
+    SequenceLabelingEvaluator,
+    TaggingScheme,
+)
+from embeddings.metric.hugging_face_metric import HuggingFaceMetric
+from embeddings.metric.seqeval_metric import SeqevalMetric
+from embeddings.metric.unit_seqeval_metric import UnitSeqevalMetric
+
+
+class SeqevalTorchMetric(Metric):
+    def __init__(
+        self,
+        class_label: ClassLabel,
+        average: str = "macro",
+        evaluation_mode: EvaluationMode = EvaluationMode.CONLL,
+        tagging_scheme: Optional[TaggingScheme] = None,
+        dist_sync_on_step: bool = False,
+    ) -> None:
+        assert average in ["macro", "micro", "weighted"]
+        super().__init__(dist_sync_on_step=dist_sync_on_step)
+        self.class_label = class_label
+        self.average = average
+        self.evaluation_mode = evaluation_mode
+        self.tagging_scheme = tagging_scheme
+        self.metric = self._get_metric()
+        self.add_state("y_pred", default=list(), dist_reduce_fx="cat")
+        self.add_state("y_true", default=list(), dist_reduce_fx="cat")
+
+    def _get_metric(
```

Comment:
changed method to static and utilized in `SeqevalTorchMetric`

Reply:
looks like class feature envy. I think it will be better to move the function outside of any class.

Reply:
well to me it does not necessary looks bad since we just use one method from another class which is static so we do not access any of its fields and we use it just once in the `__init__`; 

```
A classic smell is a method that seems more interested in a class other than the one it is in. The most common focus of the envy is the data.
``` 

if we were just consider above statement then it would be okay to implement it like this